tests/python/contrib/test_vitis_ai_codegen.py

```diff
@@ -0,0 +1,203 @@
+            subgraph_mod["main"] = funcnode
+            with tvm.transform.PassContext(opt_level=3, config={'target_':'DPUCADX8G'}):
+                fcompile(subgraph_mod["main"])
+
+
+def test_add():
+    shape = (10, 10)
+    x = relay.var('x', shape=shape)
+    y = x + x
+    func = relay.Function([x], y)
+    _construct_model(func)
+
+def test_relu():
+    shape = (10, 10)
+    x = relay.var('x', shape=shape)
+    y = relay.nn.relu(x)
+    func = relay.Function([x], y)
+    _construct_model(func)
+
+def test_conv2d():
+    x = relay.var('x', shape=(1, 3, 224, 224))
+    w = relay.const(np.zeros((16, 3, 3, 3), dtype='float32'))
+    y = relay.nn.conv2d(x, w, strides=[2, 2], padding=[1, 1, 1, 1], kernel_size=[3, 3])
+    func = relay.Function([x], y)
+    params = {}
+    params["x"] = np.zeros((16, 3, 3, 3), dtype='float32')
+    _construct_model(func, params)
+
+
+def test_global_avg_pool2d():
+    shape = (10, 10, 10, 10)
+    x = relay.var('x', shape=shape)
+    y = relay.nn.global_avg_pool2d(x)
+    func = relay.Function([x], y)
+    _construct_model(func)
+
+def test_annotate():
+    """Test annotation with Vitis-AI DP (DPUCADX8G)"""
+    def partition():
+        data = relay.var("data", relay.TensorType((1, 3, 224, 224), "float32"))
+        weight = relay.var("weight", relay.TensorType((16, 3, 3, 3), "float32"))
+        bn_gamma = relay.var("bn_gamma", relay.TensorType((16, ), "float32"))
+        bn_beta = relay.var("bn_beta", relay.TensorType((16, ), "float32"))
+        bn_mmean = relay.var("bn_mean", relay.TensorType((16, ), "float32"))
+        bn_mvar = relay.var("bn_var", relay.TensorType((16, ), "float32"))
+
+        conv = relay.nn.conv2d(
+            data=data,
+            weight=weight,
+            kernel_size=(3, 3),
+            channels=16,
+            padding=(1, 1))
+        bn_output = relay.nn.batch_norm(conv, bn_gamma, bn_beta, bn_mmean,
+                                        bn_mvar)
+
+        func = relay.Function([data, weight, bn_gamma, bn_beta, bn_mmean,
+                               bn_mvar], bn_output.astuple())
+        mod = tvm.IRModule()
+        mod["main"] = func
+        params = {}
+        params["weight"] = np.random.rand(16, 3, 3, 3).astype('float32')
+        params["bn_gamma"] = np.random.rand(16).astype('float32')
+        params["bn_beta"] = np.random.rand(16).astype('float32')
+        params["bn_mean"] = np.random.rand(16).astype('float32')
+        params["bn_var"] = np.random.rand(16).astype('float32')
+        mod = annotation(mod, params, "DPUCADX8G")
+
+        opt_pass = tvm.transform.Sequential([
+            transform.InferType(),
+            transform.PartitionGraph(),
+            transform.SimplifyInference(),
+            transform.FoldConstant(),
+            transform.AlterOpLayout(),
+        ])
+
+        with tvm.transform.PassContext(opt_level=3):
+            mod = opt_pass(mod)
+
+        return mod
+
+    def expected():
```

Comment:
It seems to me that you're expecting two subgreaphs (functions)? But your previous comment says the flow currently only supports one subgraph in a model.